src/store/module/todo.js

```diff
@@ -0,0 +1,85 @@
+import storage from "../storage.js";
+import Todo from "../../domain/Todo.js";
+import { FILTER } from "../../utils/FILTER.js";
+
+const mapToTodo = (items) => {
+  return items.map(
+    ({ _id, contents, isCompleted }) => new Todo(_id, contents, isCompleted)
+  );
+};
+
+const initFilter = () => {
+  if (location.hash.includes(FILTER.ACTIVE)) {
+    return FILTER.ACTIVE;
+  } else if (location.hash.includes(FILTER.COMPLETED)) {
+    return FILTER.COMPLETED;
+  }
+  return FILTER.ALL;
+};
+
+const todo = (() => {
+  const items = mapToTodo(storage.getStorage());
+  let nextId = items[items.length - 1]._id + 1;
+  let filter = initFilter();
+
+  const getNewId = () => {
+    return nextId++;
+  };
+
+  const addItem = (item) => {
+    items.push(item);
+    storage.setStorage(items);
+  };
+
+  const findItemById = (id) => {
+    return items.find((item) => item.isSameId(id));
+  };
+
+  const toggleItem = (id) => {
+    const item = findItemById(id);
+    item.toggle();
+    storage.setStorage(items);
+  };
+
+  const deleteItem = (id) => {
+    const deleteItemIndex = items.findIndex((item) => item.isSameId(id));
+    items.splice(deleteItemIndex, 1);
+    storage.setStorage(items);
+  };
```

Comment:
```suggestion
  const deleteItem = (targetId) => {
    const newItmes = items.filter((item) => item.id !== targetId);
    storage.setStorage(newItems);
  };
```

변수명을 명확히하고 상황에 맞는 적절한 메서드를 활용하면 더 가독성 높은 코드가 될거 같습니다!

Reply:
현재의 프로그램은 초기에 한번만 localStorage에서 데이터를 호출하고 이후에는 저장만 하는 방식을 취하고 있습니다.
splice가 아닌 filter로 바꾸게 된다면 원본 배열은 변하지 않기 때문에 todo 에서 가지고 있는 items가 의미없게 되겠네요.🤔

api로 치환하고 외부 데이터를 호출한다고 생각했을 때, 외부에서 변경될 가능성이 있으니 내부 데이터 보다는 외부 데이터를 우선 시 하는 게 맞는 것 같기도 하군요.
todo의 items를 지우고 getItems() 함수를 생성해 매번 데이터를 호출하도록 변경했습니다.

좋은 리뷰 감사드려요 😄

Reply:
아하 그렇군요! 제가 코드를 꼼꼼히 확인하지 못했네요! todo에서 가지고 있는 items에 대한 부분은 생각 못 했던거 같습니다!

좋은 의견 감사드립니다!!👍👍👍